pageserver/src/walingest.rs

```diff
@@ -1355,8 +1355,10 @@ impl WalIngest {
             // Note: The multixact members can wrap around, even within one WAL record.
             offset = offset.wrapping_add(n_this_page as u32);
         }
-        if xlrec.mid >= self.checkpoint.nextMulti {
-            self.checkpoint.nextMulti = xlrec.mid + 1;
+        if xlrec.mid.wrapping_sub(self.checkpoint.nextMulti) as i32 >= 0 {
+            let next_mid =
+                std::cmp::max(xlrec.mid.wrapping_add(1), pg_constants::FIRST_MULTIXACT_ID);
+            self.checkpoint.nextMulti = next_mid;
             self.checkpoint_modified = true;
         }
         if xlrec.moff + xlrec.nmembers > self.checkpoint.nextMultiOffset {
```

Comment:
This has the same bug

Reply:
Sorry, I thought that offset can not wraparound:(
Fixed.